codecov.yml

```diff
@@ -13,3 +13,10 @@ coverage:
     patch: false
     changes: false
 comment: off
+
+ignore:
+  - "**/setup.py"
+  - "**/tests/*"
+  - "**/test_*.py"
```

Comment:
The linked issue makes a good point about flushing through tests to make sure you don't i.e., duplicate test names.

Is there a plan to have a "separate context" for unit tests only as suggested there? While that may seem like a good idea, it is also a fair bit of extra information (overload for reviewer?) if it can be done & perhaps even more maintenance burden since each extra CI component will always need some occasional attention.

I'd be +1 for just keeping the tests in the main report, but don't really mind either way. Ignoring the setup stuff seems sensible.

Reply:
Indeed, the initial idea was to see purely python results as an experiment (dips down about 5%). However, including test files for coverage to check the tests seems a bit indirect to me even though I agree about the point raised in the linked issue. 

There might be valid reasons to skip tests or leaving code out without changing the overall coverage. Not to mention it boosts the coverage result too much. But having yet another coverage specifically for tests seems to be bloating the test cycle for not much of a benefit. We are closely following the new code additions and the old code is there for ages and (I guess) no one is following the code coverage that closely.

To be honest I don't know enough about pytest to form an opinion about this and possible consequences. I just wanted to kickstart the discussion. If we decide to not to care then probably we do nothing about it.